ion-schema/src/nfa.rs

```diff
@@ -267,12 +269,12 @@ impl NfaEvaluation {
     // this method iterates through elements to satisfy minimum required occurrences for given transition
     // It will return false if an invalid Ion value is found which doesn't satisfy minimum occurrence requirement for given transition
     // Otherwise it will return true
-    fn evaluate_transition_to_self(
+    fn evaluate_transition_to_self<'a, T: Iterator<Item = &'a Element>>(
```

Comment:
Food for thought: you could create a new trait for this to avoid having to spell out the bounds each time.
```rust
// Make an empty extension trait called `ElementIter`...
trait ElementIter<'a>: Iterator<Item = &'a Element> {}
impl<'a, T> ElementIter<'a> for T where T: Iterator<Item = &'a Element> {}

// ...and then replace this:
fn evaluate_transition_to_self<'a, T: Iterator<Item = &'a Element>> (...) {...}

// ...with this:
fn evaluate_transition_to_self<'a, T: ElementIter<'a>>(...) {...}
```

Reply:
Yeah, I guess I could. TBH, though, I'm hoping to do a fairly complete rewrite the NFA to get better error messages like in https://github.com/amazon-ion/ion-schema-kotlin/pull/228